### PR TITLE
Transf elas

### PR DESCRIPTION
**Pull request by EliLazarus**
Fix transformation elasticities

### CODE REVIEW

**Comment by EliLazarus** [comment]
(Other than test coverage), this passes all the CET with different CES elasticity values cases in I have in there, and doesn't break any other tests! 
Next steps for this branch before a merge: 
- Clean up all the remnant comments and print lines from my experimentation
- Check, is it ok that the input shareparamaters are using total inputs in the denominator? 
- Confirm it is ok to rebuild at every solve
- Potentially get clarity on the different expression versions from different substitution elasticities in the implicit constraints

**Comment by EliLazarus** [comment]
At this point it's working and cleaned up, resolves #81
- There is an issue with replicating the set of ShareParameter every solve!/build! (this change includes new build with every solve!). 
There may be more efficient or elegant ways to do what I implemented, review for that would be great.
[This fix is robust to all cases so far - I'm crossing my fingers I have broad test coverage for cases, but I will be double-checking, nevertheless, I think this is ready for merge].

File: src/build/build_implicitconstraints.jl
Comment (davidanthoff) [inline src/build/build_implicitconstraints.jl:157]: I think this entire diff doesn't actually change anything? `swap_our_param_with_val(demand.commodity)` is the same as `get_commodity_benchmark(demand.commodity)`, so we again just have a fraction here that cancels out to 1?

File: src/build/build_implicitconstraints.jl
Comment (davidanthoff) [inline src/build/build_implicitconstraints.jl:209]: Does this change anything? Don't think so, right? Would be easier to review if you could keep formatting changes out of the PR.

File: src/build/build_implicitconstraints.jl
Comment (davidanthoff) [inline src/build/build_implicitconstraints.jl:229]: This is the only substantial change, really? Replacing the division with a multiplication? Was that just the bug why things didn't work before?

**Review by davidanthoff — COMMENTED** [review]
Just some random comments, the next step should be that we merge my new PR into this branch here, and then I'll review again.

File: examples/Testofparameters.jl
Comment (davidanthoff) [inline examples/Testofparameters.jl:17]: Why is there so much code that is commented out? Should this file even committed at all, I think it would be much cleaner if a PR like this one here just had changes to `src` and the `test` directory. This looks more like a scratch file?

File: examples/example1.jl
Comment (davidanthoff) [inline examples/example1.jl:34]: Should this be here?

File: examples/example2.ipynb
Comment (davidanthoff) [inline examples/example2.ipynb:401]: Why are the changes to this file part of this PR? They are unrelated, so they shouldn't come along this PR here.

File: src/model.jl
Comment (davidanthoff) [inline src/model.jl:623]: What does that mean? In what way? Is this related to transfer elasticities in some way?

File: test/test_Benchmark-Theta.jl
Comment (davidanthoff) [inline test/test_Benchmark-Theta.jl:164]: The entire file is not properly formatted, all the indenting seems inconsistent? Makes it hard to read.

File: test/test_twobytwo_macro.jl
Comment (davidanthoff) [inline test/test_twobytwo_macro.jl:138]: Why is there so much commented code?

**Review by davidanthoff — APPROVED** [review]
@EliLazarus let's merge this as is, if this all looks good to you. We can fix any remaining things in follow up PRs.

File: src/build/build_implicitconstraints.jl
Comment (EliLazarus) [reply src/build/build_implicitconstraints.jl:166]: No, this isn't actually a change, just re-order of operations for ease of reading/interpretation.

File: test/test_Benchmark-Theta.jl
Comment (EliLazarus) [reply test/test_Benchmark-Theta.jl:164]: oops

File: examples/example2.ipynb
Comment (EliLazarus) [reply examples/example2.ipynb:401]: Happened automatically somehow, I guess I accidentally staged it.

File: examples/example1.jl
Comment (EliLazarus) [reply examples/example1.jl:34]: No. More scratch/experimenting. I'll delete it.

File: src/model.jl
Comment (EliLazarus) [reply src/model.jl:623]: Unrelated :/  . It just means that one can't re-set the consumer value, it's always calculated as the sum of endowments in the Demand.

**Pull request merged**
